fideslib/core/config.py

```diff
@@ -0,0 +1,272 @@
+    AES_GCM_NONCE_LENGTH: int = 12
+    APP_ENCRYPTION_KEY: str
+    DRP_JWT_SECRET: str
+
+    @validator("APP_ENCRYPTION_KEY")
+    def validate_encryption_key_length(  # pylint: disable=E0213, R0201
+        cls, v: Optional[str], values: Dict[str, str]
+    ) -> Optional[str]:
+        """Validate the encryption key is exactly 32 bytes"""
+        if v is None or len(v.encode(values.get("ENCODING", "UTF-8"))) != 32:
+            raise ValueError("APP_ENCRYPTION_KEY value must be exactly 32 bytes long")
+        return v
+
+    CORS_ORIGINS: List[AnyHttpUrl] = []
+
+    @validator("CORS_ORIGINS", pre=True)
+    def assemble_cors_origins(  # pylint: disable=E0213, R0201
+        cls, v: Union[str, List[str]]
+    ) -> Union[List[str], str]:
+        """Return a list of valid origins for CORS requests"""
+        if isinstance(v, str) and not v.startswith("["):
+            return [i.strip() for i in v.split(",")]
+        if isinstance(v, (list, str)):
+            return v
+        raise ValueError(v)
+
+    ENCODING: str = "UTF-8"
+
+    # OAuth
+    OAUTH_ROOT_CLIENT_ID: str
+    OAUTH_ROOT_CLIENT_SECRET: str
+    OAUTH_ROOT_CLIENT_SECRET_HASH: Optional[Tuple]
+    OAUTH_ACCESS_TOKEN_EXPIRE_MINUTES: int = 60 * 24 * 8
+    OAUTH_CLIENT_ID_LENGTH_BYTES = 16
+    OAUTH_CLIENT_SECRET_LENGTH_BYTES = 16
+
+    @validator("OAUTH_ROOT_CLIENT_SECRET_HASH", pre=True)
+    def assemble_root_access_token(  # pylint: disable=E0213, R0201
+        cls, _: Optional[str], values: Dict[str, str]
+    ) -> Tuple:
+        """Returns a hashed value of the root access key.
+
+        This is hashed as it is not wise to return a plaintext for of the
+        root credential anywhere in the system.
+        """
+        value = values.get("OAUTH_ROOT_CLIENT_SECRET")
+        if not value:
+            raise MissingConfig(
+                "OAUTH_ROOT_CLIENT_SECRET is required", SecuritySettings
+            )
+
+        encoding = values["ENCODING"]
+
+        salt = bcrypt.gensalt()
+        hashed_client_id = hashlib.sha512(value.encode(encoding) + salt).hexdigest()
+        return hashed_client_id, salt
+
+    class Config:  # pylint: disable=C0115
+        env_prefix = "FIDES__SECURITY__"
+
+
+class FidesConfig(FidesSettings):
+    """Configuration variables for the FastAPI project."""
+
+    database: DatabaseSettings
+    security: SecuritySettings
+
+    is_test_mode: bool = os.getenv("TESTING") == "True"
+    hot_reloading: bool = os.getenv("FIDES__HOT_RELOAD") == "True"
+    dev_mode: bool = os.getenv("FIDES__DEV_MODE") == "True"
+
+    class Config:  # pylint: disable=C0115
+        case_sensitive = True
+
+    logger.warning(
+        "Startup configuration: reloading = %s, dev_mode = %s", hot_reloading, dev_mode
+    )
+    logger.warning(
+        'Startup configuration: pii logging = %s == "True"}',
+        os.getenv("FIDES__LOG_PII"),
+    )
+
+
+def load_file(file_names: Union[List[Path], List[str], Path, str]) -> str:
```

Comment:
I originally had it this way and changed it since if it gets passed it it will always only be one file. If everyone is fine with passing a list even though there will alway only be one element in it that is fine with me.